### PR TITLE
bpo-40275: Remove duplicate import of locale module

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue40275

### CODE REVIEW

**Review by vstinner — CHANGES_REQUESTED** [review]
This PR goes somehow against https://bugs.python.org/issue40275 goal. I would prefer to move the top-level "import locale" into skip_if_buggy_ucrt_strfptime().

**Comment by shihai1991** [comment]
> This PR goes somehow against https://bugs.python.org/issue40275 goal. I would prefer to move the top-level "import locale" into skip_if_buggy_ucrt_strfptime().

make sense, I updated this PR.

**Comment by shihai1991** [comment]
Oh, the gate of macos failed randomly (no generated log file)

**Comment by serhiy-storchaka** [comment]
It does not help. `locale` is imported by `gettext` which is imported by `argparse` which is imported by `unittest`.

For the purpose of bpo-40275 the changes should reduce the list of imported modules.

**Comment by vstinner** [comment]
> It does not help. locale is imported by gettext which is imported by argparse which is imported by unittest.

Maybe we could attempt to make locale import lazy in gettext, or make gettext import lazy in argparse.

Or just give up because locale is less heavy than asyncio for example :-)

**Comment by shihai1991** [comment]
> > It does not help. locale is imported by gettext which is imported by argparse which is imported by unittest.
> 
> Maybe we could attempt to make locale import lazy in gettext, or make gettext import lazy in argparse.
> 
> Or just give up because locale is less heavy than asyncio for example :-)

I try to make local import lazy in gettext. I don't make gettext import lazy in argparse, because  much function will call gettext in argpase(it's looks untidy).

File: Lib/gettext.py [hunk 1]
```diff
@@ -46,7 +46,6 @@
 #   find this format documented anywhere.
 
 
-import locale
 import os
```
Comment (vstinner) [inline Lib/gettext.py]: Please write a separated PR for gettext, and keep this PR for test.support.

Comment (shihai1991) [reply Lib/gettext.py]: OK, I created #19905  to do this job.

**Pull request merged**